### PR TITLE
how-to-use-virtio-mem-with-kata.md: Update doc to make it clear

### PR DESCRIPTION
**Pull request by teawater**
Update this howto because the virtio-mem support of kata, qemu and Linux
was updated.

Fixes: #1845

Signed-off-by: Hui Zhu <teawater@antfin.com>

### CODE REVIEW

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 1]
```diff
@@ -13,26 +13,20 @@ Kata Containers with `virtio-mem` supports memory resize.
...
+The `virtio-mem` config of the x86_64 Kata Linux kernel is open.
+Enable `virtio-mem` as follows:
+```
+$ sudo sed -i -e 's/^#enable_virtio_mem.*$/enable_virtio_mem = true/g' /etc/kata-containers/configuration.toml
 ```
 
-Enable `virtio-mem` with following line in the Kata Containers configuration:
-```toml
-enable_virtio_mem = true
+### With others
```
Comment (fidencio) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:23]:
```suggestion
### With other architectures
```

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 2]
```diff
@@ -41,13 +35,35 @@ Use following command to enable memory overcommitment of a Linux kernel.  Becaus
 $ echo 1 | sudo tee /proc/sys/vm/overcommit_memory
 ```
 
-Use following command start a Kata Container.
+Use following command start a Kata Container with containerd.
```
Comment (fidencio) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:38]:
```suggestion
Use following command to start a Kata Container.
```

This shouldn't be specific to containerd, right?

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 3]
```diff
@@ -41,13 +35,35 @@ Use following command to enable memory overcommitment of a Linux kernel.  Becaus
...
+- top
+EOF
+$ sudo crictl pull $image
+$ podid=$(sudo crictl runp $pod_yaml)
+$ cid=$(sudo crictl create $podid $container_yaml $pod_yaml)
+$ sudo crictl start $cid
 ```
 
-Use following command set the memory size of test to default_memory + 512m.
+Use following command set the container memory limit to 2g and the memory size of VM to default_memory + 2g.
```
Comment (fidencio) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:61]:
```suggestion
Use the following command to set the container memory limit to 2g and the memory size of the VM to its default_memory + 2g.
```

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 4]
```diff
@@ -41,13 +35,35 @@ Use following command to enable memory overcommitment of a Linux kernel.  Becaus
...
 ```
-$ docker update -m 512m --memory-swap -1 test
+$ sudo crictl update --memory $((2*1024*1024*1024)) $cid
 ```
 
+Use following command set the container memory limit to 1g and the memory size of VM to default_memory + 1g.
```
Comment (fidencio) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:66]:
```suggestion
Use the following command to set the container memory limit to 1g and the memory size of the VM to its default_memory + 1g.
```

**Review by fidencio — COMMENTED** [review]
@teawater, I've made a few suggestions and raised one question.
This should work independently of the CRI runtime used (CRI-O or containerd), right?

**Comment by teawater** [comment]
> @teawater, I've made a few suggestions and raised one question.
> This should work independently of the CRI runtime used (CRI-O or containerd), right?

Yes, I think it should support any of them.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 1 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Updated.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 2 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Updated.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 3 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Updated.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 4 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Updated.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 5]
```diff
@@ -13,26 +13,20 @@ Kata Containers with `virtio-mem` supports memory resize.
...
-Set Linux and the QEMU that support `virtio-mem` with following line in the Kata Containers QEMU configuration `configuration-qemu.toml`:
-```toml
-[hypervisor.qemu]
-path = "qemu-dir"
-kernel = "vmlinux-dir"
+### With x86_64
+The `virtio-mem` config of the x86_64 Kata Linux kernel is open.
+Enable `virtio-mem` as follows:
+```
+$ sudo sed -i -e 's/^#enable_virtio_mem.*$/enable_virtio_mem = true/g' /etc/kata-containers/configuration.toml
```
Comment (devimc) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:20]: hmmm I think this is QEMU specific, isn't it?  also I don't have `/etc/kata-containers/` in my system, what should I do? :wink:

**Review by devimc — COMMENTED** [review]
thanks @teawater - just one comment

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 5 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md:23]: Updated to make them clear.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 6]
```diff
@@ -13,26 +13,23 @@ Kata Containers with `virtio-mem` supports memory resize.
 
 ## Requisites
 
-Kata Containers with `virtio-mem` requires Linux and the QEMU that support `virtio-mem`.
-The Linux kernel and QEMU upstream version still not support `virtio-mem`.  @davidhildenbrand is working on them.
-Please use following unofficial version of the Linux kernel and QEMU that support `virtio-mem` with Kata Containers.
+Kata Containers just support `virtio-mem` with QEMU.
```
Comment (GabyCT) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:16]: s/support/supports

File: docs/how-to/how-to-use-virtio-mem-with-kata.md [hunk 7]
```diff
@@ -41,13 +38,35 @@ Use following command to enable memory overcommitment of a Linux kernel.  Becaus
 $ echo 1 | sudo tee /proc/sys/vm/overcommit_memory
 ```
 
-Use following command start a Kata Container.
+Use following command to start a Kata Container.
 ```
-$ docker run --rm -it --runtime=kata --name test busybox
+$ pod_yaml=pod.yaml
+$ container_yaml=${REPORT_DIR}/container.yaml
+$ image="busybox:latest"
```
Comment (GabyCT) [inline docs/how-to/how-to-use-virtio-mem-with-kata.md:45]: maybe change it to `quay.io/prometheus/busybox:latest`?

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 6 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Fixed.

File: docs/how-to/how-to-use-virtio-mem-with-kata.md (hunk 7 above)
Comment (teawater) [reply docs/how-to/how-to-use-virtio-mem-with-kata.md]: Updated.

**Review by Tim-Zhang — APPROVED** [review]

**Review by lifupan — APPROVED** [review]

**Pull request merged**